ivy/functional/frontends/tensorflow/raw_ops.py

```diff
@@ -228,13 +228,26 @@ def MatMul(*, a, b, transpose_a=False, transpose_b=False, name="MatMul"):
     return ivy.matmul(a, b, transpose_a=transpose_a, transpose_b=transpose_b)
 
 
+@to_ivy_arrays_and_back
 def MatrixInverse(*, input, adjoint=False, name="MatrixInverse"):
     return ivy.inv(input, adjoint=adjoint)
 
 
 MatrixDeterminant = tf_frontend.linalg.det
 
 
+@to_ivy_arrays_and_back
+def MatrixDiag(*, diagonal, name="MatrixDiag"):
+    last_dim = diagonal.shape[-1]
+
+    return ivy.astype(
+        ivy.expand_dims(diagonal, axis=-1)
```

Comment:
Wouldn't it be more efficient if we make use of `ivy.zeros` and `ivy.scatter_nd` to update the array in place? this would make it more efficient for backends that supports `native_out`, we can generate the diag indicies using a list comprehension, e.g. `[(i,i) for i in diag]` and use it along with `scatter_nd`. What do you think?